From a73896d5be3cb49127703a0bc2cf067fe7f0de7e Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Tue, 22 Sep 2020 12:24:17 -0700
Subject: [PATCH 1641/1851] v4l: xilinx: axis-switch: Fix warnings for UNINIT

commit 7236b955a3892d849e198e88f765f3415af3c22c from
https://github.com/Xilinx/linux-xlnx.git

This patch fix the warnings for the UNINIT type.

The actual warnings are:

For xvsw_parse_of():
Event var_decl: Declaring variable "routing_mode" without initializer.
Event uninit_use: Using uninitialized value "routing_mode".

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-axis-switch.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-axis-switch.c b/drivers/media/platform/xilinx/xilinx-axis-switch.c
index 120beb76af7b..c5e4319ed19f 100644
--- a/drivers/media/platform/xilinx/xilinx-axis-switch.c
+++ b/drivers/media/platform/xilinx/xilinx-axis-switch.c
@@ -386,7 +386,7 @@ static int xvsw_parse_of(struct xvswitch_device *xvsw)
 	struct device_node *ports;
 	struct device_node *port;
 	unsigned int nports = 0;
-	u32 routing_mode;
+	u32 routing_mode = 0;
 	int ret;
 
 	ret = of_property_read_u32(node, "xlnx,num-si-slots", &xvsw->nsinks);
-- 
2.31.1

